cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -1207,7 +1288,105 @@ void search(raft::resources const& res,
             raft::device_matrix_view<float, int64_t, raft::row_major> distances,
             const cuvs::neighbors::filtering::base_filter& sample_filter =
               cuvs::neighbors::filtering::none_sample_filter{});
+/**
+ * @brief Search ANN using the composite cagra index.
+ *
+ * See the [cagra::build](#cagra::build) documentation for a usage example.
+ *
+ * @param[in] res raft resources
+ * @param[in] params configure the search
+ * @param[in] index composite cagra index
+ * @param[in] queries a device matrix view to a row-major matrix [n_queries, index->dim()]
+ * @param[out] neighbors a device matrix view to the indices of the neighbors in the source dataset
+ * [n_queries, k]
+ * @param[out] distances a device matrix view to the distances to the selected neighbors [n_queries,
+ * k]
+ * @param[in] sample_filter an optional device filter function object that greenlights samples
+ * for a given query. (none_sample_filter for no filtering)
+ */
+void search(raft::resources const& res,
```

Comment:
Hi @cjnolet @achirkin , I had to add a new search for `composite_index`, the root cause is the [`cuve::neighbors::index` ](https://github.com/rapidsai/cuvs/blob/branch-25.04/cpp/include/cuvs/neighbors/common.hpp#L87)is not designed to be an abstract class, which makes many pathways impossible,  like inheriting from `index` can avoid to declare a `composible_index` and a new search API. I was trying to refactor it, but I found it was used widely in different algo codes; maybe we can make it in the long term.

Reply:
I think it's okay that we don't require a strict class hierarchy initially, so long as we have proper overloads so that from a user's perspective, they can call `auto index = cuvs::neighbors::cagra::merge(...)` and get back a proper object from which they can later call `cuvs::neighbors::cagra::search(..., index,..)`